### PR TITLE
Discover base url

### PR DESCRIPTION
**Pull request by dogeared**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -112,7 +112,7 @@
         <!-- Test Dependencies: Only required for testing when building the SDK.  Not required by SDK users at runtime: -->
         <groovy.version>1.8.8</groovy.version>
         <logback.version>1.1.2</logback.version>
-        <easymock.version>3.1</easymock.version>
+        <easymock.version>3.4</easymock.version>
```
Comment (dogeared) [inline pom.xml]: FYI, easymock 3.4 supports the `partialMock` construct used in the test in this PR. Earlier version do not have partial mocks.

File: extensions/spring/stormpath-spring/src/main/java/com/stormpath/spring/config/AbstractStormpathConfiguration.java [hunk 2]
```diff
@@ -35,6 +35,8 @@
 
 import java.io.IOException;
 import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
```
Comment (mzumbadoate) [inline extensions/spring/stormpath-spring/src/main/java/com/stormpath/spring/config/AbstractStormpathConfiguration.java]: Are Matcher and Pattern used in this class?

File: impl/src/main/java/com/stormpath/sdk/impl/client/DefaultClientBuilder.java [hunk 3]
```diff
@@ -122,4 +124,17 @@ public ClientBuilder setBaseUrl(String baseUrl) {
         return this;
     }
 
+    @Override
+    public ClientBuilder setDefaultBaseUrl(String applicationHref) {
+        if (applicationHref == null) {
+            throw new IllegalArgumentException("applicationHref argument cannot be null.");
+        }
+        Pattern pattern = Pattern.compile("(.*)://([^/]*)/([^/]*)/.*");
```
Comment (mzumbadoate) [inline impl/src/main/java/com/stormpath/sdk/impl/client/DefaultClientBuilder.java]: can this regex be a constant? hopefully in a global scope?

**Comment by dogeared** [comment]
Shouldn't be part of the public API. Needs to be reworked.

**Comment by lhazlewood** [comment]
Another issue closed with an orphan branch - what's going on here?

**Comment by dogeared** [comment]
@lhazlewood - This was identified as a "nice to have" feature and did not have an issue or milestone associated with it. Going forward, all PRs will be associated with an issue and a milestone.

I've created an issue and referenced the branch as a good starting point. It's #493

**Pull request closed without merging**